docs/DEVELOPMENT-GUIDE.md

```diff
@@ -113,11 +113,12 @@ if (s2n_do_something(with_something_else) < 0) {
 }
 ```
 
-is so common that utils/s2n_safety.h provides two macros:
+is so common that utils/s2n_safety.h provides three macros:
 
 ```c
-#define GUARD( x )      if ( (x) < 0 ) return -1
-#define GUARD_PTR( x )  if ( (x) < 0 ) return NULL
+#define GUARD( x )         if ( (x) < 0 ) return -1
+#define GUARD_STRICT( x )  if ( (x) != 0 ) return -1
```

Comment:
do we need `GUARD_STRICT_PTR`?

Reply:
That's a very good point! I haven't found a use case for it yet but it might be that some OpenSSL function or the like has this non-zero error functionality, and is called in a function returning a pointer. I'll add it, and no harm no foul if we never have to use it.